### ISSUE TITLE
sensors display a value from another sensor

### ISSUE THREAD

**Issue by bredita** [5e/streamdeck-hwinfo-plugin#6]
TLDR; 
It seems like any sensor with a single digit Label works. 
Any sensor with two or three digits in Label gets the value from the last digit in its label:
Label 121 -> Value1
Label 112 -> Value2
etc

_I think the problem is related to this, in sensor.ts:_

![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/1892b618-d2da-4164-bceb-2ea7336a41fe)

FYI: The largest "index number" in my VSB registry folder is Label/Value169:

**Long explanation:**

I tried again, adding buttons for the core frequencies for all 16 cores, 
Renamed the sensors to Core # Clock (in case the perf numbers change over time).
And the three "Physical Memory" values to "Physical RAM" to make registry search easier.
Made sure no sensors are disabled.
Added them in the plugin, and it looks like this. 

(I basically got the same result with "Shared memory support" enabled and disabled.
And with "Disable drive scan" enabled and SAS/SMART/ATA disabled.)

The 6 first cores show a clock speed. I think the correct one. 
core 7-9 shows the sensor values for "Physical memory used".
core 10-15 seem OK, they show a clock speed, but I'm not sure...

![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/3aee8dc0-645a-4640-adce-e4dbc09ee243)

The registry says:

HWINFO64/VSB folder in registry:

Label0 = Physical RAM Used
Label1 = Physical RAM Available 
Label2 = Physical RAM Load
Label3 = Core 0 Clock
Label4 = Core 1 Clock
Label5 = Core 2 Clock
Label6 = Core 3 Clock
Label7 = Core 4 Clock
Label8 = Core 5 Clock
Label9 = Core 6 Clock

Label10 = Core 7 Clock
Label11 = Core 8 Clock
Label12 = Core 9 clock
Label13 = Core 10 clock

Label20 = Core 0 T1 Usage
Label21 = Core 1 T0 Usage
Label22 = Core 1 T1 Usage
Label23 = Core 2 T0 Usage

Label120 = PMIC High Temperature DIMM 1
Label121 = PMIC Over Voltage DIMM 1
Label122 = PMIC Under Voltage DIMM 1
Label123 = SPD HUB Temperature DIMM 2

I suspect that all the ones that fail have more than one digit sensor Labels/Values:
I added Label1-4 to my streamdeck, and any other sensor that has the same last digit to the right of it:

![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/cce941c5-1d7c-4c7d-bbda-9640046077f1)

As you can see, all sensors with a label that ends in 0 show the same data.
Same with the ones that end in 1, 2 and 3.

**Other things I checked that probably are unimportant:**

I installed the 8gadgetpack and the HWInfoMmonitor plugin (since this is the "reason" for adding the values to the registry, according to hwinfo documentation). Tried both version 1.1 and 2.0 of this 8gadgetpack plugin and they both require "Shared memory support" in HWinfo. They both work though.

**Clocks folder in registry HWINFO64/VSB:**

Clock6: 
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/0fa341ff-0e08-45c1-84bd-19a2c771acdd)

Clock7:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/dc15a40d-30e1-43cb-b189-1b37e9831626)

Clock8:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/31209efa-f863-4a39-b203-fd76419fecc6)

Clock9:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/5a57e140-63d8-40db-94c8-a9e3becc6323)

_To me it looks like the data in the VSB folder is consistent?_

Other registry entries that mention the same cores, probably nothing important here:

**Clocks folders in HWINFO64/Sensors/E0002000_0:**

Clock6:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/0327fca7-9c9d-4ded-b810-4c9bb1d2050c)

Clock7:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/01cecd00-96b9-43ee-aea1-c25ab8179eb1)

(There is no clock 8-9 in this folder)

**Clocks folders in HWINFO64/Sensors/F0000300_0:**

Clock6:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/d25000f1-b5ed-47bb-b91b-70696fa72f91)

Clock7:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/6360e997-f2d1-4eea-9b84-dead5133160b)

Clock8:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/4e14e7dd-c689-464b-b7b6-a12d9e13de5f)

Clock9:
![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/14e74321-a7d1-4847-920f-74febfa6a968)

(all clocks 0-15have an entry in this folder)

(HWINFO64/Sensors/F0000501_0 also have a couple of Clock folders, Clock37 and Clock64.)

I just noticed you published version 1.0.2, so I uninstalled 1.0.1 and tried 1.0.2, but it didn't fix this issue.

**Comment by 5e** [comment]
You are completely right about the way of getting the index is wrong, no idea how I missed that, will fix this.

**Comment by 5e** [comment]
Hi, 

https://filebin.net/exctln3y5vechbtq/com.5e.hwinfo-reader.streamDeckPlugin

If you would like to give it a go, uninstall the plugin and let me know if this version now works @bredita

**Comment by bredita** [comment]
Perfect. Thank you.

![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/6dc958b8-33fa-4d3a-bb33-b49cf7348f4c)

![image](https://github.com/5e/streamdeck-hwinfo-plugin/assets/4368888/a1bb7f0d-0e0a-46e4-aaab-dd61df9314c5)

**Comment by bredita** [comment]
the one from pastebin works just fine now. Version number is still 1.0.2 according to the settings/plugins tab, guess that will update when elgato pushes it to the store?

**Comment by 5e** [comment]
Yep, will be updated when it’s on sent to the store but just going to add 1 more thing to this update before shipping it
